src/parser/statement.js

```diff
@@ -652,62 +652,72 @@ pp.parseClassBody = function (node) {
       continue;
     }
 
-    let method = this.startNode();
+    let elementNode = this.startNode();
 
     // steal the decorators if there are any
     if (decorators.length) {
-      method.decorators = decorators;
+      elementNode.decorators = decorators;
       decorators = [];
     }
 
+    if (this.match(tt.hash)) { // Private property
+        if (!this.hasPlugin("classPrivateProperties")) {
+          return this.raise(this.state.pos, `Unexpected character '${tt.hash.label}'`);
+        }
+        this.next();
```

Comment:
Will this ignore whitespace? In the current spec, whitespace between the # and the IdentifierName is not allowed (though Waldemar proposed that it be allowed).